### PR TITLE
add labels for notebook so that gcp credentials will be injected by webhook

### LINKED ISSUE #2770: update notebook controller so that notebook pod has proper annotation for admission webhook
see #2641

### PR DESCRIPTION
**Pull request by lluunn**
Fix #2770

/cc @jlewi @zabbasi

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2853)

### CODE REVIEW

File: components/notebook-controller/pkg/controller/notebook/notebook_controller.go [hunk 1]
```diff
@@ -283,6 +284,14 @@ func generateStatefulSet(instance *v1alpha1.Notebook) *appsv1.StatefulSet {
 			},
 		},
 	}
+
+	// Inject GCP credentials
+	if os.Getenv("GCP_CREDENTIALS") != "" {
```
Comment (jlewi) [inline components/notebook-controller/pkg/controller/notebook/notebook_controller.go]: How about generalizing this?

Take an environment variable that is a comma separated list of labels to apply to the pods.

Comment (lluunn) [reply components/notebook-controller/pkg/controller/notebook/notebook_controller.go]: done

**Comment by abhi-g** [comment]
@lluunn is this waiting for another review from @jlewi

**Comment by lluunn** [comment]
yes, thanks

**Comment by r2d4** [comment]
We also need the environment variable `GOOGLE_APPLICATION_CREDENTIALS` to be set and point to the mounted secret to have parity with the previous Kubeflow jupyter deployment.

**Pull request merged**